test/src/api/commands/client/testSetNetworkConditions.js

```diff
@@ -0,0 +1,78 @@
+const assert = require('assert');
+const CommandGlobals = require('../../../../lib/globals/commands.js');
+const MockServer = require('../../../../lib/mockserver.js');
+const Nightwatch = require('../../../../lib/nightwatch.js');
+
+describe('.setNetworkConditions()', function () {
+  beforeEach(function (done) {
+    CommandGlobals.beforeEach.call(this, done);
+  });
+
+  afterEach(function (done) {
+    CommandGlobals.afterEach.call(this, done);
+  });
+
+  it('browser.setNetworkConditions()', function (done) {
+    MockServer.addMock(
+      {
+        url: '/session',
+        response: {
+          value: {
+            sessionId: '13521-10219-202',
+            capabilities: {
+              browserName: 'chrome',
+              browserVersion: '92.0'
+            }
+          }
+        },
+        method: 'POST',
+        statusCode: 201
+      },
+      true
+    );
+
+    Nightwatch.initW3CClient({
+      desiredCapabilities: {
+        browserName: 'chrome',
+        'goog:chromeOptions': {}
+      }
+    }).then((client) => {
+      client.transport.driver.setNetworkConditions = function (spec) {
+
+        return Promise.resolve();
```

Comment:
can we assert on `specs` received over here?

Reply:
Updated tests in latest commit